Common/src/geometry/dual_grid/CPoint.cpp

```diff
@@ -26,430 +26,155 @@
  */
 
 #include "../../../include/geometry/dual_grid/CPoint.hpp"
+#include "../../../include/CConfig.hpp"
+#include "../../../include/omp_structure.hpp"
 
-CPoint::CPoint(unsigned short val_nDim, unsigned long val_globalindex, CConfig *config) : CDualGrid(val_nDim) {
+CPoint::CPoint(unsigned long npoint, unsigned long ndim) : nDim(ndim) {
 
```

Comment:
Significant change 3: There is a "minimal constructor" that only allocated the fields needed to read the mesh file (when geometry_aux is built).